ci(test-sheet-13): musl build failing on alpine:latest (3 | 3.23 | 3.23.0) related to `mbsnrtowcs`

Used `alpine:3.22` to fix the CI issue with https://github.com/liquidaty/zsv/pull/521/commits/6c65741b1e590a43fbbe96a4a4a7ef4712f1b7b8.

Dependabot PR: https://github.com/liquidaty/zsv/pull/521
CI failure: https://github.com/liquidaty/zsv/actions/runs/20220169091/job/58040626531

<details>

<summary>Output of failing `test-sheet-13` (Unable to convert to wide chars)</summary>

```text
test-sheet-13: 
Incorrect output:
����S���H�T$XdH+%(
                  Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                                             Una
ble to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                        Unable to convert to wid
e chars: 1��A9��'����S���H�T$XdH+%(
                                   Unable to convert to wide chars: 1��A9��'����
S���H�T$XdH+%(
              Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                                         Unable
to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                    Unable to convert to wide ch
ars: 1��A9��'����S���H�T$XdH+%(
                               Unable to convert to wide chars: 1��A9��'����S���
H�T$XdH+%(
          Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                                     Unable to c
onvert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                Unable to convert to wide chars:
 1��A9��'����S���H�T$XdH+%(
                           Unable to convert to wide chars: 1��A9��'����S���H�T$
XdH+%(
      Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                                 Unable to conve
rt to wide chars: 1��A9��'����S���H�T$XdH+%(
                                            Unable to convert to wide chars: 1��
A9��'����S���H�T$XdH+%(
                       Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+
%(
  Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                             Unable to convert t
o wide chars: 1��A9��'����S���H�T$XdH+%(
                                        Unable to convert to wide chars: 1��A9��
'����S���H�T$XdH+%(
                   Unable to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                                              Un
able to convert to wide chars: 1��A9��'����S���H�T$XdH+%(
                                                         Unable to convert to wi
de chars: 1��A9��'����S���H�T$XdH+%(

Expected output:
make[2]: *** [Makefile:880: test-sheet-13] Error 1
make[1]: *** [Makefile:616: test-standalone] Error 2
make: *** [Makefile:59: test] Error 2
Key(s)                    Action                    Description
q                         quit                      Exit the application
<esc>                     escape                    Leave the current view or
^                         first                     Jump to the first column
$                         last                      Jump to the last column
<shift><left>             first                     Jump to the first column
<shift><right>            last                      Jump to the last column
k                         up                        Move up one row
j                         down                      Move down one row
h                         left                      Move left one column
l                         right                     Move right one column
<up>                      up                        Move up one row
<down>                    down                      Move down one row
<left>                    left                      Move left one column
<right>                   right                     Move right one column
<ctrl>d                   pagedown                  Move down one page
<ctrl>u                   pageup                    Move up one page
<page up>                 pagedown                  Move down one page
<page down>               pageup                    Move up one page
g g                       top                       Jump to the first row
G                         bottom                    Jump to the last row
/                         find                      Set a search term and jum
n                         next                      Jump to the next search r
|                         gotocolumn                Go to column
e                         open                      Open another CSV file
f                         filter                    Filter by specified text
F                         filtercol                 Filter by specified text
:                         subcommand                Editor subcommand
?                         help                      Display a list of actions
^J                        <Enter>                   Follow hyperlink (if any)
^M                        <Enter>                   Follow hyperlink (if any)
v                         pivot                     Group rows by the column
V                         pivotexpr                 Group rows with group-by
                          where                     Filter by sql expression
                          errors                    Show errors (if any)
                          errors-clear              Clear any/all errors



<esc> to exit help Key(s)
cmp /__w/zsv/zsv/tmp/test-sheet-13.out expected/test-sheet-13.out
/__w/zsv/zsv/tmp/test-sheet-13.out expected/test-sheet-13.out differ: char 1, line 1
test-sheet-13: Failed!
make[2]: Leaving directory '/__w/zsv/zsv/app/test'
make[1]: Leaving directory '/__w/zsv/zsv/app'
Error: Process completed with exit code 2.
```

</details>

Related code:

https://github.com/liquidaty/zsv/blob/90ba5871800249b7361d7e7cacd2f128471e6883/app/sheet.c#L919C19-L924

Related `mbsnrtowcs` musl libc threads:

- Mailing List:
  - https://www.openwall.com/lists/musl/2025/04/29/1
  - https://www.openwall.com/lists/musl/2025/04/29/4
- Commit (2025-05-05):
  - https://git.musl-libc.org/cgit/musl/commit/?id=23febbd3c5d6b4e367faee4f39b8e8c3b95ab588

The latest musl libc release is [v1.2.5](https://git.musl-libc.org/cgit/musl/tag/?h=v1.2.5) published on 2024-03-01.
Most probably, a new release i.e. v1.2.6 will fix this issue.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -595,7 +595,7 @@ jobs:
   ci-musl:
     needs: [tag, clang-format, cppcheck, shellcheck]
     runs-on: ubuntu-latest
-    container: alpine:3.22
+    container: alpine:latest
     timeout-minutes: 15
 
     outputs:
@@ -607,7 +607,10 @@ jobs:
     steps:
     - name: Set up dependencies
       shell: sh
-      run: apk add bash gcc make musl-dev ncurses-dev ncurses-static tmux file sqlite curl zip wget tar git
+      run: |
+        apk update
+        apk add bash make musl-dev ncurses-dev ncurses-static tmux file sqlite curl zip wget tar git
+        apk add gcc=14.2.0-r6 --repository=https://dl-cdn.alpinelinux.org/alpine/v3.22/main
 
     - name: Checkout
       uses: actions/checkout@v7

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,10 +4,19 @@ LABEL maintainer="Liquidaty"
 LABEL url="https://github.com/liquidaty/zsv"
 LABEL org.opencontainers.image.description="zsv: tabular data swiss-army knife CLI + world's fastest (simd) CSV parser"
 
-RUN apk add bash gcc make musl-dev ncurses-dev ncurses-static tmux file sqlite curl zip
+RUN apk update && apk add bash make musl-dev ncurses-dev ncurses-static tmux file sqlite curl zip git
+
+# GCC 15 onwards causes failure due to the default strict aliasing rules in the compiler.
+# GCC 14 is the last version that works with zsv, so we need to install it from the v3.22 repository. 
+RUN apk add gcc=14.2.0-r6 --repository=https://dl-cdn.alpinelinux.org/alpine/v3.22/main
 
 WORKDIR /zsv
 COPY . .
+
+# For some reason, the diff file is not being applied correctly in the docker build context.
+# git apply on local docker build fails to resolve the full path for parallel tests.
+# So, we need to remove the prefix to make paths relative in the diff file before applying it.
+RUN sed "s|app/test/parallel/||" -i app/test/parallel/chunk_break.diff
 
 RUN mkdir /usr/local/etc
 

